src/dragonball/src/tracer.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2023 Alibaba Cloud
+// Copyright (c) 2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+
+use anyhow::Result;
+use log::info;
+use tracing::subscriber::DefaultGuard;
+use tracing::{span::EnteredSpan, subscriber::NoSubscriber, trace_span, Subscriber};
+
+const DEFAULT_JAEGER_URL: &str = "http://localhost:14268/api/traces";
+
+/// Configuration information for tracing feature.
+#[derive(Clone)]
+pub struct TraceInfo {
+    /// Sandbox id.
+    pub sid: String,
+    /// subscriber from runtime.
+    pub subscriber: Arc<dyn Subscriber + Send + Sync>,
+}
+
+impl Debug for TraceInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TraceConfigInfo")
+            .field("sid", &self.sid)
+            .finish()
+    }
+}
+
+impl PartialEq for TraceInfo {
```

Comment:
Why do you need to implement PartialEq

Reply:
`TraceInfo` is used in `VmmAction::SetHypervisorTracing(TraceInfo)` and `VmmAction` implements the PartialEq. This is also the reason why I added the `sid` field to `TraceInfo`.